homework_7/hw_fun.py

```diff
@@ -0,0 +1,80 @@
+import sys
+
+
+# Sequential map
+
+def sequential_map(*args) -> list:
+    args = [*args]
+    cont = args.pop()
+    for fun in args:
+        cont = list(map(fun, cont))
+    return cont
+
+
+# Consensus filter
+
+def consensus_filter(*args) -> list:
+    args = [*args]
+    cont = args.pop()
+    for fun in args:
+        cont = list(filter(fun, cont))
+    return cont
+
+
+# Conditional reduce
+
+def conditional_reduce(fun_1, fun_2, cont) -> int:
+    filt_cont = iter(filter(fun_1, cont))
+    val = next(filt_cont)
+    for el in filt_cont:
+        val = fun_2(val, el)
+    return val
+
+
+# Functional chain
+
+def func_chain(*functions):
+    funcs = [*functions][::-1]
+
+    def composite_function(*args):
+        arguments = list(args)
+        while funcs:
+            arguments = list(map(funcs.pop(), arguments))
+        match len(arguments):
+            case 1:
+                return (lambda x: x)(*arguments)
+            case _:
+                return arguments
```

Comment:
В данном случае, эта конструкция только усложняет чтение, по сути ты просто переписала логику `if`
```suggestion
        if len(arguments) == 1:
            return (lambda x: x)(*arguments)
        else:
            return arguments
```